### PR TITLE
Feature/advanced botany infu

### PR DESCRIPTION
**Pull request by EnderProyects**
Update recipes, buildscript, add catalyst to NEI, and changet ingot to plates in recipes

### CODE REVIEW

File: dependencies.gradle [hunk 1]
```diff
@@ -2,11 +2,12 @@ dependencies {
     api('com.github.GTNewHorizons:Botania:1.9.16-GTNH:dev')
     api('com.github.GTNewHorizons:Botanic-horizons:1.0.15-GTNH:dev')
     api('thaumcraft:Thaumcraft:1.7.10-4.2.3.5:dev')
-    api('com.github.GTNewHorizons:GT5-Unofficial:5.09.42.71:dev')
+    //api('com.github.GTNewHorizons:GT5-Unofficial:5.09.42.71:dev')
```
Comment (boubou19) [inline dependencies.gradle]: remove, version control is here if you need to add it back.

File: src/main/java/ab/common/lib/register/RecipeListAB.java [hunk 2]
```diff
@@ -1,5 +1,8 @@
 package ab.common.lib.register;
 
+import static gregtech.api.enums.Mods.ForbiddenMagic;
+import static gregtech.api.util.GT_ModHandler.getModItem;
+
```
Comment (boubou19) [inline src/main/java/ab/common/lib/register/RecipeListAB.java]: this wont work, as GT is yeeted from the deps

File: src/main/java/ab/common/lib/register/RecipeListAB.java [hunk 3]
```diff
@@ -24,6 +27,7 @@
 import thaumcraft.api.research.ResearchCategories;
 import thaumcraft.api.research.ResearchItem;
 import thaumcraft.api.research.ResearchPage;
+import thaumcraft.api.wands.*;
```
Comment (boubou19) [inline src/main/java/ab/common/lib/register/RecipeListAB.java]: wildcard imports are dangerous, please configure your ide to not use them, explicit imports are way better.

**Review by boubou19 — CHANGES_REQUESTED** [review]
did not looked at recipe changes

File: src/main/java/ab/common/lib/register/RecipeListAB.java (hunk 2 above)
Comment (EnderProyects) [reply src/main/java/ab/common/lib/register/RecipeListAB.java]: This was the code that @chochem give to me to solve a variable, idk if it will work without that

Comment (boubou19) [reply src/main/java/ab/common/lib/register/RecipeListAB.java]: copying code without understanding it is a very bad practice.

Comment (chochem) [reply src/main/java/ab/common/lib/register/RecipeListAB.java]: that was before we started talking about removing gt as a dep obviously

Comment (EnderProyects) [reply src/main/java/ab/common/lib/register/RecipeListAB.java]: Bad practice: "Asking for help for a problem that I dint understand and that somebody help me with that"

**Comment by EnderProyects** [comment]
@boubou19 Could you review this when you have time? 👉👈

File: src/main/java/ab/common/core/proxy/CommonProxy.java [hunk 4]
```diff
@@ -10,6 +10,7 @@
 import ab.common.lib.register.FlowerRegister;
 import ab.common.lib.register.ItemListAB;
 import ab.common.minetweaker.MineTweakerConfig;
+import ab.nei.*;
```
Comment (boubou19) [inline src/main/java/ab/common/core/proxy/CommonProxy.java]: wildcard import bad, please configure properly your IDE to avoid them.

File: src/main/java/ab/common/lib/register/RecipeListAB.java [hunk 5]
```diff
@@ -34,6 +34,7 @@
 import vazkii.botania.api.lexicon.LexiconPage;
 import vazkii.botania.api.recipe.RecipePetals;
 import vazkii.botania.common.Botania;
+import vazkii.botania.common.block.*;
```
Comment (boubou19) [inline src/main/java/ab/common/lib/register/RecipeListAB.java]: wildcard import

File: src/main/java/ab/common/lib/register/RecipeListAB.java [hunk 6]
```diff
@@ -55,6 +56,7 @@
 import ab.utils.*;
 import ab.utils.LocalizationManager;
 import ab.utils.OreDict;
+import cpw.mods.fml.common.registry.*;
```
Comment (boubou19) [inline src/main/java/ab/common/lib/register/RecipeListAB.java]: wildcard import

File: src/main/java/ab/common/lib/register/RecipeListAB.java [hunk 7]
```diff
@@ -55,6 +56,7 @@
 import ab.utils.*;
```
Comment (boubou19) [inline src/main/java/ab/common/lib/register/RecipeListAB.java]: wildcard import

**Review by boubou19 — DISMISSED** [review]

**Comment by boubou19** [comment]
didn't paid attention to what you changed gameplay wise, but other than wildcard imports, it looks ok

File: src/main/java/ab/common/core/proxy/CommonProxy.java [hunk 8]
```diff
@@ -34,6 +35,7 @@ public void preInit(FMLPreInitializationEvent event) {
     public void init(FMLInitializationEvent event) {
         NetworkRegistry.INSTANCE.registerGuiHandler(AdvancedBotany.instance, new GuiHandler());
         NetworkHandler.registerPackets();
+        IMCForNEI.IMCSender();
```
Comment (combusterf) [inline src/main/java/ab/common/core/proxy/CommonProxy.java:38]: This looks unrelated to the PR description, please elaborate

File: src/main/java/ab/common/core/proxy/CommonProxy.java (hunk 4 above)
Comment (EnderProyects) [reply src/main/java/ab/common/core/proxy/CommonProxy.java]: The internal wildcard imports are used because I use all the contents of that file / package

File: src/main/java/ab/common/lib/register/RecipeListAB.java [hunk 9]
```diff
@@ -405,9 +413,8 @@ public static void setupCrafting() {
                 new ItemStack(BlockListAB.blockManaCrystalCube),
                 new Object[] { "TST", "GWG", "DMD", Character.valueOf('D'), new ItemStack(ModBlocks.dreamwood),
                         Character.valueOf('W'), new ItemStack(ModItems.twigWand, 1, 32767), Character.valueOf('G'),
-                        new ItemStack(ModBlocks.manaGlass), Character.valueOf('M'),
-                        new ItemStack(ModItems.manaResource), Character.valueOf('T'),
-                        new ItemStack(ModItems.manaResource, 1, 18), Character.valueOf('S'),
+                        new ItemStack(ModBlocks.manaGlass), Character.valueOf('M'), new ItemStack(ModBlocks.storage),
+                        Character.valueOf('T'), new ItemStack(ModItems.manaResource, 1, 18), Character.valueOf('S'),
```
Comment (combusterf) [inline src/main/java/ab/common/lib/register/RecipeListAB.java]: This looks a lot like symptomatic decompiled code and it makes things unnecessarily hard to read. While I don't expect you to fix the entire codebase at once, I would recommend that changed recipes follow idiomatic java. Compare with:

```
GameRegistry.addShapedOreRecipe(
  new ItemStack(BlockListAB.blockManaCrystalCube),
  "TST", "GWG", "DMD",
  'D', new ItemStack(ModBlocks.dreamwood),
  'W', new ItemStack(ModItems.twigWand, 1, 32767), // is the magic number necessary?
  'G', new ItemStack(ModBlocks.manaGlass),
  'M', "blockManaSteel", // not sure if its the right one, but must be oredict for GT compat
  'T', new ItemStack(ModItems.manaResource, 1, 18), // prefer oredict
  'S', new ItemStack(ModItems.spark)
);
```

File: src/main/java/ab/common/lib/register/RecipeListAB.java [hunk 10]
```diff
@@ -791,6 +761,90 @@ public static void setupCrafting() {
...
+                new ItemStack[] { new ItemStack(ModItems.elementiumChest), new ItemStack(ModItems.laputaShard, 1, 15),
+                        new ItemStack(ModBlocks.storage, 1, 2), new ItemStack(ItemListAB.itemABResource, 1, 5),
+                        new ItemStack(ItemListAB.itemNebulaRing), new ItemStack(ItemListAB.itemABResource, 1, 5),
+                        new ItemStack(BlockListAB.blockABStorage, 1, 0),
+                        new ItemStack((Item) Item.itemRegistry.getObject("Thaumcraft:ItemAmuletRunic"), 1, 1),
+                        new ItemStack(ModItems.terrasteelChest),
+                        new ItemStack((Item) Item.itemRegistry.getObject("Thaumcraft:ItemAmuletRunic"), 1, 1),
+                        new ItemStack(BlockListAB.blockABStorage, 1, 0), new ItemStack(ItemListAB.itemABResource, 1, 5),
+                        new ItemStack(ItemListAB.itemNebulaRing), new ItemStack(ItemListAB.itemABResource, 1, 5),
+                        new ItemStack(ModBlocks.storage, 1, 2), new ItemStack(ModItems.laputaShard, 1, 15),
```
Comment (combusterf) [inline src/main/java/ab/common/lib/register/RecipeListAB.java]: ModBlocks.storage must be replaced with oredictionary for GT material compatibility.

**Review by combusterf — CHANGES_REQUESTED** [review]

File: src/main/java/ab/common/core/proxy/CommonProxy.java (hunk 8 above)
Comment (EnderProyects) [reply src/main/java/ab/common/core/proxy/CommonProxy.java:38]: This is for NEI compatibility, so you can see catalyst icons in the recipes, it was going to go into another pr, but yesterday I have free time, so I added it

File: src/main/java/ab/common/lib/register/RecipeListAB.java (hunk 9 above)
Comment (miozune) [reply src/main/java/ab/common/lib/register/RecipeListAB.java]: 32767 == `OreDictionary.WILDCARD_VALUE`

File: src/main/java/ab/common/lib/register/RecipeListAB.java (hunk 10 above)
Comment (EnderProyects) [reply src/main/java/ab/common/lib/register/RecipeListAB.java]: The modBlocks.storage are only from botania, so no need of oredic it, if you mean to place something like this is okay for me:
 new ItemStack(ModBlocks.storage, 1, Constants.STORAGE_META_MANASTEELBLOCK),

File: src/main/java/ab/common/lib/register/RecipeListAB.java (hunk 9 above)
Comment (EnderProyects) [reply src/main/java/ab/common/lib/register/RecipeListAB.java]: @combusterf I need to change this? I mean if I make new recipes ( that for the time I dont think so), I will use this method, also the magic number is because the recipe can use every single wand of the forest

File: src/main/java/ab/common/lib/register/RecipeListAB.java (hunk 10 above)
Comment (EnderProyects) [reply src/main/java/ab/common/lib/register/RecipeListAB.java]: Done

File: src/main/java/ab/common/lib/register/RecipeListAB.java [hunk 11]
```diff
@@ -486,8 +490,9 @@ public static void setupCrafting() {
         // Sphere of Attraction recipe
         GameRegistry.addShapedRecipe(
                 new ItemStack(ItemListAB.itemAntigravityCharm),
-                new Object[] { " G ", "GDG", "ERE", Character.valueOf('E'), new ItemStack(ModItems.manaResource, 1, 7),
-                        Character.valueOf('G'), new ItemStack(ModBlocks.elfGlass), Character.valueOf('D'),
+                new Object[] { " G ", "GDG", "ERE", Character.valueOf('E'),
+                        OreDict.preference("plateElvenElementium", OreDict.ELEMENTIUM_PLATE), Character.valueOf('G'),
+                        new ItemStack(ModBlocks.elfGlass), Character.valueOf('D'),
```
Comment (combusterf) [inline src/main/java/ab/common/lib/register/RecipeListAB.java]: Points from the previous review that have not been addressed:
1: Use oredictionary recipes instead of basic recipes (look up `ShapedOreRecipe`) so that the ore dictionary is actually used as intended instead of pretending. (mandatory)
2: remove `new Object[] {}`, the java compiler generates that for you
3: remove references to `Character.valueOf`, the java compiler also generates those for you
4: remove anything that pins an oredictionary reference to a single item (`OreDict.preference(...)`, `getOres(...).get(0)`) and use the string directly in the recipe. (mandatory)
5: make sure the recipe letter and the matching ingredient are on the same line of code.

**Review by combusterf — CHANGES_REQUESTED** [review]
Improper Oredictionary implementation (attempts have been made but at least it's not yet functional as such)

File: src/main/java/ab/common/core/proxy/CommonProxy.java (hunk 8 above)
Comment (combusterf) [reply src/main/java/ab/common/core/proxy/CommonProxy.java:38]: I see the PR description has been updated :+1:

File: src/main/java/ab/common/lib/register/RecipeListAB.java (hunk 10 above)
Comment (combusterf) [reply src/main/java/ab/common/lib/register/RecipeListAB.java]: GT materials mod generates blocks, plates and all other things, and while it *currently* doesn't create doubles where the native botania version exists, it has done so for plenty of other mods and therefore that's not a good assumption to make.

File: src/main/java/ab/common/lib/register/RecipeListAB.java (hunk 11 above)
Comment (EnderProyects) [reply src/main/java/ab/common/lib/register/RecipeListAB.java]: image
Something like this?

**Review by combusterf — APPROVED** [review]
Some things still don't use oredictionary. It's probably a really good idea to improve `AdvancedBotanyAPI.registerAdvancedPlateRecipe` so that it doesn't cause issues in the future.

**Comment by Dream-Master** [comment]
@EnderProyects are the Missing thaumcraft pages added to the book?
@Alastors
